app/controllers/api/v1/cookie_and_jwt_api/protocol_subscriptions_controller.rb

```diff
@@ -25,6 +39,29 @@ def set_protocol_subscription
 
           render(status: :not_found, json: 'Protocol subscription met dat ID niet gevonden')
         end
+
+        def start_date
+          return start_time if protocol_subscription_create_params[:start_date].blank?
+
+          Time.zone.parse(protocol_subscription_create_params[:start_date])
+        end
+
+        def start_time
+          return Time.zone.now if no_start_time_given
+
+          next_midnight = TimeTools.increase_by_duration(Time.zone.now.beginning_of_day, 1.day)
+          TimeTools.increase_by_duration(next_midnight, protocol_subscription_create_params[:start_time].to_i)
+        end
+
+        def no_start_time_given
+          protocol_subscription_create_params[:start_time].blank? ||
+            (protocol_subscription_create_params[:start_time].to_i.zero? &&
+              protocol_subscription_create_params[:start_time] != '0')
+        end
+
+        def protocol_subscription_create_params
+          params.permit(:protocol_name, :start_date, :start_time)
+        end
```

Comment:
on my device the indentation seems weird on the next lines

Reply:
I reformatted it, but it didn't change anything. Weird.